### PR TITLE
feat: add wasmsnip pass

### LINKED ISSUE #1: Add support for wasm-snip
https://github.com/rustwasm/wasm-snip

I'm imagining exposing options like:

* `wasm-snip.snip-rust-panicking-code: Boolean`: Whether to remove the rust ``std::panicking` code.

* `wasm-snip.snip-rust-fmt-code: Boolean`: Whether to remove the rust `std::fmt` code.

* `wasm-snip.functions: Array<String|RegExp>` Exact function names or regexp patterns to snip.

### PR DESCRIPTION
**Pull request by xtuc**
Closes #1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,6 +15,9 @@ yarn add rust-plugin
 | `debug` | Enable debug mode |
 | `wasmopt.level` | configure `wasm-opt` optimization level (default `z`) |
 | `wasmopt.required` | throw if the binary is not available (default `false`: warn) |
+| `wasmopt.snipRustPanickingCode` | Snip Rust's `std::panicking` and `core::panicking` code.(default `true`) |
+| `wasmopt.snipRustFmtCode` | Snip Rust's `std::fmt` and `core::fmt` code (default `true`). |
+| `wasmopt.functions` | Snip any function that matches the given regular expression or a string (default empty). |
```
Comment (fitzgen) [inline README.md]: These should be `wasmsnip.*`, right?

Comment (xtuc) [reply README.md]: yes, good catch

Comment (fitzgen) [reply README.md]: Oh, also: I don't think we should make snipping the panicking and fmt code on by default. It is pretty disastrous if you didn't intentionally do that, and you'll see your code just start trapping and throwing exceptions that are hard to debug :-p

Comment (xtuc) [reply README.md]: You're seeing an outdated diff, see a373939

Comment (fitzgen) [reply README.md]: Great :)

**Pull request merged**